### ISSUE TITLE
Numbered list

### ISSUE THREAD

**Issue by SofiaDubbelve** [sebgroup/design-library#122]
We have a need of a numbered list. We can not find that in bootstrap today, see link https://sebgroup.github.io/bootstrap/list-group the closest to what we need is the ordered list, but we will not need the states. We want everything to be visible at the same time. A page that has this, however color and size might be up for debate, is https://seb.se/foretag/bli-foretagskund/starta-eget-foretag-steg-for-steg and I found that there is a component in pattern library that has this https://designlibrary.sebgroup.com/components/upcoming/public-web-components/ordered-bullet-list/ but it is green here, and not in bootstrap. We want such a component in bootstrap. Could it be the boxes in color #333333 ?

**Comment by ulde01** [comment]
What do you think, @hjalmers? Is this a new component or a twist of the ones that Sofia mentions?

**Comment by SofiaDubbelve** [comment]
![image](https://user-images.githubusercontent.com/46597224/81296220-dc4d9e80-9071-11ea-9d10-23a1eea405f1.png)
![image](https://user-images.githubusercontent.com/46597224/81296835-c8566c80-9072-11ea-92ae-1abadeed3bbf.png)

**Comment by SofiaDubbelve** [comment]
32 x 32, to make room for 2 numbers if it  is a very long list! much to do!

**Comment by hjalmers** [comment]
> What do you think, @hjalmers? Is this a new component or a twist of the ones that Sofia mentions?

In bootstrap the base class or block is called `list-group` and today we have one modifier for ordered lists called `list-group-ordered`. E.g. to get an ordered list you'd do this today:

```html
<ol class="list-group list-group-ordered">
    <li class="list-group-item">Fist list item</li>
    ...
</ol>
```

### List modifiers
We have other modifiers for other lists like, bullets and check lists. This could of course be another one perhaps ´list-group-ordered-highlighted´ or it could just be a modifier for the list group item. Each list group item (bullet in the list) can have a modifier applied to it give it a different meaning. 

The default one is just `list-group-item` which a applies to all lists (bullet, check, flush and ordered).
We rarely add modifiers to list-group-items unless we need to convey a state or a feeling for a particular item, e.g. it's active, disabled, has error, step is completed etc. like we do when the list is used in the wizard component. 

The black background is the color we've chosen to denote that an item is active, which is very similar to this "new" modifier with two major differences:

* The active version i smaller (not 2 rem/ 32 px), more on that later
* The active version also adds a light grey background to the list item

**Active version**
![image](https://user-images.githubusercontent.com/1632755/81377996-cccc6500-9106-11ea-999b-834fb393d837.png)

**The code:**
```html
<ol class="list-group list-group-ordered">
    <li class="list-group-item active">Hey look at me, I'm active!</li>
</ol>
```

Bootstrap also provides a number of utility classes that can be used with any element to modify it without adding a new version or new code, one of them that could be used in this case is the background modifier `bg-*` and for this particular case we could use either `bg-white` (if it's against a white background) or `bg-transparent` to achieve this:

**Active version with bg modifier**
![image](https://user-images.githubusercontent.com/1632755/81378670-f5a12a00-9107-11ea-9acb-83a1fdc23d52.png)

**The code:**
```html
<ol class="list-group list-group-ordered">
    <li class="list-group-item bg-transparent active">Hey look at me, I'm active with a transparent background!</li>
    <li class="list-group-item bg-white active">Hey look at me, I'm active with a white background!</li>
</ol>
```

### Different sizes?
Now we´re only off on the size requirement. In my opinion the nicest would be to introduce a new modifier for list groups something like `list-group-lg` if we want to have larger version (this could apply to check lists and other lists as well and not just the ordered list). However if we're just introducing this bigger version to hold two digits, I think we should consider other options. The current version is actually big enough to display two digits (see previous example) so it's not a major issue, but we could always increase the default size a bit get more padding when using long lists. Then again I think long lists are an anti pattern and that's it's a better to divide them into sub lists with group headers if they get too long, having lists with two digits or more is something that we should try to avoid if possible in my opinion.

### Okay
**A long list*
1. Duis pulvinar diam nec ex tristique tempor.
2. Fusce nec leo dignissim, feugiat mauris id, lacinia sapien.
3. Morbi consequat urna id sollicitudin malesuada.
4. Maecenas venenatis tellus in nunc dictum porttitor.
5. Suspendisse feugiat libero sed tempus pretium.
6. Nu kör vi svenska.
7. Idag skiner solen ute.
8. Det var nära att jag cyklade till jobbet idag.
9. Pärltomater är de bästa tomaterna om du frågar mig.
10. Jag har tre skärmar framför mig när jag sitter på kontoret.
11. Okay, switching to english now.
12. Another english bullet

### Better
**Latin**
1. Duis pulvinar diam nec ex tristique tempor.
2. Fusce nec leo dignissim, feugiat mauris id, lacinia sapien.
3. Morbi consequat urna id sollicitudin malesuada.
4. Maecenas venenatis tellus in nunc dictum porttitor.
5. Suspendisse feugiat libero sed tempus pretium.

**Swedish**
1. Nu kör vi svenska.
2. Idag skiner solen ute.
3. Det var nära att jag cyklade till jobbet idag.
4. Pärltomater är de bästa tomaterna om du frågar mig.
5. Jag har tre skärmar framför mig när jag sitter på kontoret.

**English**
1. Okay, switching to english now.
2. Another english bullet

### How should lists look by default?
If this is the default appearance we want for ordered lists I'd advice against using too many sub modifiers i.e. it should be simple and easy to get it right and you shouldn't have to add a lot of extra classes to get the right look and feel. If that's the case, perhaps we should change the default list from grey like it is today?

**Default version**
![image](https://user-images.githubusercontent.com/1632755/81379272-15851d80-9109-11ea-9bd1-dae532c941a4.png)

**The code:**
```html
<ol class="list-group list-group-ordered">
    <li class="list-group-item">Hey look at me, I'm the default list and I'm grey!</li>
    <li class="list-group-item">Hey look at me, I'm the default list and I'm grey!</li>
    <li class="list-group-item">Hey look at me, I'm the default list and I'm grey!</li>
</ol>
```

If not, I think it will start getting confusing not just for user but for us as well. When to use a normal grey list and when to use the black version? Why is this list black and that one grey, and in one list black means something and in another something else. Not to mention maintaining dark mode versions of each modification if that's something we want to support in the future. I'm a big fan of [KISS-principle](https://en.wikipedia.org/wiki/KISS_principle) `Keep It Stupid Simple` so if I could decide I would go for just one ordered list and either keep it grey like it it is or revise it. For edge cases or special circumstances it's always possible to use other modifiers like `bg-*`, `text-*` etc. to override default behavior without adding more css or creating new components just for the sake of it.

**Comment by ghost** [comment]
Hi, me and Robert had a talk about the ordered list. We agreed on this:
**To be consistent with our way to mark a component with a status (with colours), like we do in the Wizard step thing. We will not use the grey box behind the number when there is no status to show. So the background box is only used on active (dark grey) and completed (green) and step that has something missed out or wrong (red). Maybe disabled as well?! (light grey). (SORRY @SofiaDubbelve) Default numbered list with no status will be only the numbers no background.**

@hjalmers Please correct me if I didn´t explain it correctly 😅

**Comment by hjalmers** [comment]
Yes, we also decided to make the background color of the active state slightly darker (grey 200 instead of gray 100) so that's it's visible even when used directly on our body background color which is grey 100 or white according to #36 and https://github.com/sebgroup/bootstrap/issues/282)

### Normal ordered list
This how the ordered list looks without the "box" background.

![image](https://user-images.githubusercontent.com/1632755/82783422-b7716d80-9e5e-11ea-9cbb-249ddf74881e.png)

### States for ordered list items
These are the states that can be added to a list item, note that I added the info state to align with how we use alerts, toasts etc.

![image](https://user-images.githubusercontent.com/1632755/82786444-ea1e6480-9e64-11ea-9cf8-07d93aa5fd33.png)
